lte/gateway/python/magma/mobilityd/subscriberdb_client.py

```diff
@@ -30,33 +30,69 @@ def get_subscriber_ip(self, sid: str) -> ipaddress.ip_address:
         if self.subscriber_client is None:
             return None
 
-        ip_addr = None
         try:
-            if '.' in sid:
-                imsi, apn_name = sid.split('.', maxsplit=1)
-            else:
-                imsi, apn_name = sid, ''
-
-            data = self.subscriber_client.GetSubscriberData(SIDUtils.to_pb(imsi))
-            if data and data.non_3gpp and data.non_3gpp.apn_config:
-                for apn_config in data.non_3gpp.apn_config:
-                    if apn_config.service_selection == '*':
-                        ip_addr = apn_config.assigned_static_ip
-                    if apn_config.service_selection == apn_name:
-                        ip_addr = apn_config.assigned_static_ip
-                        break
-
-                if ip_addr is not None:
-                    return ipaddress.ip_address(ip_addr)
-            return None
+            ip_addr, vlan = self._find_ip_and_vlan(sid)
+            return ip_addr
 
         except ValueError:
-            logging.warning("Invalid sid or ip %s: [%s]", sid, ip_addr)
+            logging.warning("Invalid data for sid %s: ", sid)
             return None
 
         except grpc.RpcError as err:
             logging.error(
-                "GetSubscriberData error[%s] %s",
+                "GetSubscriberData while reading static ip, error[%s] %s",
                 err.code(),
                 err.details())
             return None
+
+    def get_subscriber_apn_vlan(self, sid: str) -> int:
+        """
+        Make RPC call to 'GetSubscriberData' method of local SubscriberDB
+        service to get assigned IP address if any.
+        TODO: Move this API to separate APN configuration service.
+        """
+        if self.subscriber_client is None:
+            return 0
+
+        try:
+            ip_addr, vlan = self._find_ip_and_vlan(sid)
+            return vlan
+
+        except ValueError:
+            logging.warning("Invalid data for sid %s: ", sid)
+            return 0
+
+        except grpc.RpcError as err:
+            logging.error(
+                "GetSubscriberData while reading vlan-id error[%s] %s",
+                err.code(),
+                err.details())
+            return 0
+
+    def _find_ip_and_vlan(self, sid: str) -> (ipaddress.ip_address, int):
```

Comment:
the first return should be `Optional[...]` unless you want to raise when the IP is not found